### PR TITLE
Feature - Add max method count

### PR DESCRIPTION
**Pull request by jaredsburrows**
@benjamin-bader Solves #146

Add max method count, fails build when the apk/aar goes over specified max

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,7 @@ dexcount {
     enableForInstantRun = false
     teamCitySlug = null
     runOnEachAssemble = true
+    maxMethodCount = 640000
```
Comment (jaredsburrows) [inline README.md]: Add appropriate documentation.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy [hunk 2]
```diff
@@ -33,8 +33,9 @@ class DexMethodCountExtension {
     private int dxTimeoutSec = 60;
     private String teamCitySlug = null
     private boolean runOnEachAssemble = true
+    int maxMethodCount
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy]: Let's rewrite this in the future to be more "groovy-like". We can cut back on alot of the Java code and remove modifiers by writing just "pure" groovy.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy [hunk 3]
```diff
@@ -66,6 +67,7 @@ class DexMethodCountTask extends DefaultTask {
             printFullTree()
             printChart()
             printTaskDiagnosticData()
+            failBuildMaxMethods()
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy]: Run this method last, so the user(developer) can see the exception after the reports have printed to STDOUT and saved to disk.

**Comment by benjamin-bader** [comment]
Thanks, Jared!

**Pull request merged**